src/python/pants/backend/jvm/tasks/coursier_resolve.py

```diff
@@ -412,8 +412,8 @@ def _load_json_result(self, conf, compile_classpath, coursier_cache_path, invali
     # Construct a map from org:name to reconciled org:name:version
     org_name_to_org_name_rev = {}
     for coord in coord_to_resolved_jars.keys():
-      (org, name, _) = coord.split(':')
-      org_name_to_org_name_rev['{}:{}'.format(org, name)] = coord
+      m2coord = M2Coordinate.from_string(coord)
+      org_name_to_org_name_rev['{}:{}'.format(m2coord.org, m2coord.name)] = coord
```

Comment:
`org_name_to_org_name_rev` was used to as catch all for a certain `groupId:artifactId` (line 447-449), I think with your change it shouldn't be needed anymore, or at least it's worth looking into why it may be still needed and get it clarified.

Reply:
cool

Reply:
It's still needed, because conflict resolution in the json doesn't always have entries for requested coordinates. Sometimes this is because pants removes them in its own version adjustments.